### PR TITLE
Add support for OpenTelemetry tracing

### PR DESCRIPTION
**Pull request by joshk**
Adds support for:
- phoenix routes
- liveview
- oban
- and ecto

Further work will need to be done for DeviceChannel bits

The default behavior is to trace everything, but I have added support for a percentage based sampler.

To use this with honeycomb you would configure it with:
```
OTLP_ENDPOINT=https://api.honeycomb.io
OTLP_AUTH_HEADER=x-honeycomb-team
OTLP_AUTH_HEADER_VALUE=your-team-id
```

It also interesting to note that Sentry is coming out with Elixir tracing support soon

I'd like to merge this in while I work more on the Oban OpenTelemetry PR: https://github.com/open-telemetry/opentelemetry-erlang-contrib/pull/400

### CODE REVIEW

**Comment by joshk** [comment]
Screenshot 2024-10-26 at 1 35 17 PM

File: lib/nerves_hub/telemetry/filtered_sampler.ex [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  require OpenTelemetry.Tracer, as: Tracer
+  require Logger
+
+  @behaviour :otel_sampler
+
+  @ignored_static_paths ~r/^\/(assets|fonts|images)\/.*/
+
+  @ignored_url_paths [
+    "/status/alive",
+    "/favicon.ico",
```
Comment (lawik) [inline lib/nerves_hub/telemetry/filtered_sampler.ex:15]: No favicon stats? Weird :D

**Review by lawik — COMMENTED** [review]
It all made rough sense to me. I haven't done OTel specifically but looks reasonable.

Recommend others weighing in as well :)

File: lib/nerves_hub/telemetry/filtered_sampler.ex (hunk 1 above)
Comment (joshk) [reply lib/nerves_hub/telemetry/filtered_sampler.ex:15]: 😆

**Review by jjcarstens — APPROVED** [review]

File: lib/nerves_hub/deployments/orchestrator.ex [hunk 2]
```diff
@@ -48,6 +49,7 @@ defmodule NervesHub.Deployments.Orchestrator do
   As devices update and reconnect, the new orchestrator is told that the update
   was successful, and the process is repeated.
   """
+  @decorate with_span("Deployments.Orchestrator.trigger_update")
```
Comment (nshoes) [inline lib/nerves_hub/deployments/orchestrator.ex:52]: I haven't seen this before, very cool 🤤

**Review by nshoes — APPROVED** [review]

**Pull request merged**